src/movies.js

```diff
@@ -1,26 +1,164 @@
+              numberOfDramaMoviesSpielberg++;
+              return array;
+            }
+      }
+    }
+    }
+    );
+  }
+  if (numberOfDramaMoviesSpielberg===0) return 0;
+  else return numberOfDramaMoviesSpielberg;
+}
+//console.log(howManyMovies(movies));
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(array) {
+  if(array.length===0) return 0;
+    else {
+      let numberOfScores=0;
+      const sumScores = array.reduce (function (accumulator, movie) {
+        if(movie.score) {
+          numberOfScores++;
+          return accumulator + movie.score;
+        } else return accumulator;
+      }, 0);
+      return (sumScores / numberOfScores).toFixed(2);
+    }
+}
+
+//console.log (scoresAverage(movies))
+
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+
+//Option 1
+
+function dramaMoviesScore(array) {
+  if(array.length===0) return 0;
+    else {
+      let nrOfDramaMovies=0;
+      let sumScoreDrama = array.reduce((accumulator, movie) => {
+            if (movie.genre.includes('Drama')) {
+              nrOfDramaMovies++;
+              return accumulator + movie.score;
+            }
+     }, 0);
+      return (sumScoreDrama / nrOfDramaMovies).toFixed(2);
+    }
+}
+//console.log (dramaMoviesScore(movies))
+
+//Option 2:
+
+// function dramaMoviesScore(array) {
+//   if(array.length===0) return 0;
+//   else {
+//     let nrOfDramaMovies=0;
+//     let sumScoreDrama = array.reduce((accumulator, movie) => {
+//         for(let i=0; i<movie.genre.length; i++) {
+//           if (movie.genre[i]==="Drama") {
+//             nrOfDramaMovies++;
+//             return accumulator + movie.score;
+//           }
+//         }
+//     }, 0);
+//     return (sumScoreDrama/nrOfDramaMovies).toFixed(2);
+//   }
+// }
+// console.log (dramaMoviesScore(movies));
+
+
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+
+//For ordering by year only:
+
+// function orderByYear(arrayMovies) {
+//   let arrayCopy = arrayMovies.slice();
+//   arrayCopy.sort((a,b) => a.year - b.year);
+//   return arrayCopy;
+// }
+//console.log(orderByYear(movies))
+
+function orderByYear(arrayMovies) {
+  let arrayCopy = arrayMovies.slice();
+  arrayCopy.sort( function (a,b) {
+    if(a.year < b.year) return -1;
+    if(a.year > b.year) return 1;
+    if(a.year === b.year) {
+      if(a.title.toUpperCase() < b.title.toUpperCase()) return -1;
+      if(a.title.toUpperCase() > b.title.toUpperCase()) return 1;
+      if(a.title.toUpperCase() === b.title.toUpperCase()) return 0;
+    }
+  });
+  return arrayCopy;
+}
+//console.log(orderByYear(movies))
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically() {}
+
+function orderAlphabetically(arrayMovies) {
+  if(arrayMovies.length===0) return 0;
```

Comment:
why would this return 0?
the subject states that the function `receives an array and returns an array`; since this returns a number, it is invalid